Kyle_Marienthal/FLASK/full_friends/server.py

```diff
@@ -0,0 +1,79 @@
+from flask import Flask, request, redirect, render_template, session, flash
+from datetime import datetime
+from mysqlconnection import MySQLConnector
+
+app = Flask(__name__)
+mysql = MySQLConnector(app, 'full_friends')
+
+import md5, os, binascii
+password = 'password'
+
+app.secret_key = 'secret_key'
+
+def validate_registration(form_data):
+    errors = []
+    if len('name') < 2:
+        errors.append('need longer name')
+    if len('username') < 2:
+        errors.append('need longer username')
+    if len('password') < 3:
+        errors.append('need longer password')
+    if 'confirm_password' != 'password':
+        errors.append('confirmation doesnt match password')
+
+
+
+# Name
+# Username
+# Password:
+# Password Confirmation
+# Date of Birth
+
+def login_validation():
+    if len('username') < 2:
+        errors.append('need longer username')
+    if len('password') < 3:
+        errors.append('need longer password')
+# Username
+# Password
+
+@app.route("/")
+def home():
+    return render_template("login_registration.html")
+
+@app.route("/process", methods = ["POST"])
+def process():
+    salt = binascii.b2a_hex(os.urandom(15))
+    hashed_pw = md5.new(request.form['password'] + salt).hexdigest()
+    query = "INSERT INTO users(name, username, password, app_date, created_at, updated_at, salt) VALUES(:name, :username, :password, :app_date, now(), now(), :salt)"
+# the data library needs to be in the same order
```

Comment:
the data dictionary does not actually have to be in the same order. Since you're using a dictionary with keys, it will place the correct values with the correct placeholders

Reply:
that makes sense.